spec/controllers/review_controller_spec.rb

```diff
@@ -137,6 +137,74 @@
       end
     end
 
+    describe "POST request change after 2i review" do
+      let(:user) { create(:user) }
+      let(:reviewer_user) { create(:user) }
+
+      let(:step_by_step_page) do
+        create(
+          :draft_step_by_step_page,
+          review_requester_id: user.uid,
+          status: "submitted_for_2i"
+        )
+      end
+
+      before do
+        step_by_step_page.update_attributes(:status => 'in_review', :reviewer_id => reviewer_user.uid)
+      end
+
+      it "can be accessed by users with GDS editor, Unreleased feature and 2i reviewer permissions" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"]
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response).to redirect_to step_by_step_page_path(step_by_step_page)
+      end
+
+      it "cannot be accessed by users with only GDS editor permissions" do
+        stub_user.permissions << "GDS Editor"
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "cannot be accessed by users with only GDS editor and Unreleased feature permissions" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature"]
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "cannot be accessed by users with only signin permissions" do
+        stub_user.permissions = %w(signin)
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        expect(response.status).to eq(403)
+      end
+
+      it "sets status to draft, removes reviewer_id and review_requester_id" do
+        stub_user.permissions = ["signin", "GDS Editor", "Unreleased feature", "2i reviewer"]
+        post :request_change_2i_review, params: { step_by_step_page_id: step_by_step_page.id }
+
+        step_by_step_page.reload
+
+        expect(step_by_step_page.status).to eq("draft")
+        expect(step_by_step_page.reviewer_id).to be_nil
+        expect(step_by_step_page.review_requester_id).to be_nil
```

Comment:
This is nice 👍 We don't seem to be resetting `reviewer_id` and `review_requester_id` where the 2i was _approved_ - should we be? It seems like it would be good housekeeping.